skrub/_agg_joiner.py

```diff
@@ -390,49 +434,115 @@ class AggTarget(TransformerMixin, BaseEstimator):
     >>> y = np.array([1, 1, 0, 0, 1, 1])
     >>> agg_target = AggTarget(
     ...     main_key="company",
-    ...     operation=["mean", "max"],
+    ...     operations=["mean", "max"],
     ... )
     >>> agg_target.fit_transform(X, y)
-       flightId  from_airport  ...  y_0_max_target y_0_mean_target
-    0         1             1  ...               1        0.666667
-    1         2             1  ...               1        0.500000
-    2         3             1  ...               1        0.500000
-    3         4             2  ...               1        0.666667
-    4         5             2  ...               1        0.666667
-    5         6             2  ...               1        1.000000
+       flightId  from_airport  ...  y_0_mean_target  y_0_max_target
+    0         1             1  ...         0.666667               1
+    1         2             1  ...         0.500000               1
+    2         3             1  ...         0.500000               1
+    3         4             2  ...         0.666667               1
+    4         5             2  ...         0.666667               1
+    5         6             2  ...         1.000000               1
     """
 
     def __init__(
         self,
         main_key,
-        operation=None,
-        suffix=None,
+        operations,
+        *,
+        suffix="_target",
     ):
         self.main_key = main_key
-        self.operation = operation
+        self.operations = operations
         self.suffix = suffix
 
+    def check_inputs(self, X, y):
+        """Check inputs before fitting.
+
+        Parameters
+        ----------
+        X : DataFrameLike
+            Input data, base table on which to left join the target.
+        y : DataFrameLike or SeriesLike or ArrayLike
+            The target to check.
+
+        Returns
+        -------
+        y_ : DataFrameLike
+            The transformed target.
+        """
+        if not sbd.is_dataframe(X):
+            raise TypeError(f"X must be a dataframe, got {type(X)}")
+
+        self._main_key = atleast_1d_or_none(self.main_key)
+        _join_utils.check_missing_columns(X, self._main_key, "'X' (the main table)")
+
+        # `y` is copied or converted to a df to be compatible with `aggregate`
+
+        # If `y` is already a dataframe
+        if sbd.is_dataframe(y):
```

Comment:
I wonder whether we should bring the following lines into a utils function or transformer, similar to `CheckInputDataFrame`, but this time `ToDataFrame`. WDYT?